// nativescript-ui-dataform/angular/dataform-directives.js
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var element_registry_1 = require("nativescript-angular/element-registry");
var _1 = require("./../");
var RadDataFormComponent = (function () {
    function RadDataFormComponent(_elementRef) {
        this._elementRef = _elementRef;
        this._dataForm = _elementRef.nativeElement;
    }
    Object.defineProperty(RadDataFormComponent.prototype, "nativeElement", {
        get: function () {
            return this._dataForm;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(RadDataFormComponent.prototype, "dataForm", {
        get: function () {
            return this._dataForm;
        },
        enumerable: true,
        configurable: true
    });
    RadDataFormComponent = __decorate([
        core_1.Component({
            selector: "RadDataForm",
            template: ""
        }),
        __param(0, core_1.Inject(core_1.ElementRef)),
        __metadata("design:paramtypes", [core_1.ElementRef])
    ], RadDataFormComponent);
    return RadDataFormComponent;
}());
exports.RadDataFormComponent = RadDataFormComponent;
var TKEntityPropertyDirective = (function () {
    function TKEntityPropertyDirective(_elementRef) {
        this._elementRef = _elementRef;
        this._entityProperty = this._elementRef.nativeElement;
    }
    Object.defineProperty(TKEntityPropertyDirective.prototype, "entityProperty", {
        get: function () {
            return this._entityProperty;
        },
        enumerable: true,
        configurable: true
    });
    TKEntityPropertyDirective = __decorate([
        core_1.Directive({
            selector: "TKEntityProperty"
        }),
        __param(0, core_1.Inject(core_1.ElementRef)),
        __metadata("design:paramtypes", [core_1.ElementRef])
    ], TKEntityPropertyDirective);
    return TKEntityPropertyDirective;
}());
exports.TKEntityPropertyDirective = TKEntityPropertyDirective;
var TKDataFormGridLayoutDirective = (function () {
    function TKDataFormGridLayoutDirective(_elementRef) {
        this._elementRef = _elementRef;
        this._dataFormGridLayout = this._elementRef.nativeElement;
    }
    Object.defineProperty(TKDataFormGridLayoutDirective.prototype, "dataFormGridLayout", {
        get: function () {
            return this._dataFormGridLayout;
        },
        enumerable: true,
        configurable: true
    });
    TKDataFormGridLayoutDirective = __decorate([
        core_1.Directive({
            selector: "TKDataFormGridLayout"
        }),
        __param(0, core_1.Inject(core_1.ElementRef)),
        __metadata("design:paramtypes", [core_1.ElementRef])
    ], TKDataFormGridLayoutDirective);
    return TKDataFormGridLayoutDirective;
}());
exports.TKDataFormGridLayoutDirective = TKDataFormGridLayoutDirective;
var TKDataFormStackLayoutDirective = (function () {
    function TKDataFormStackLayoutDirective(_elementRef) {
        this._elementRef = _elementRef;
        this._dataFormStackLayout = this._elementRef.nativeElement;
    }
    Object.defineProperty(TKDataFormStackLayoutDirective.prototype, "dataFormStackLayout", {
        get: function () {
            return this._dataFormStackLayout;
        },
        enumerable: true,
        configurable: true
    });
    TKDataFormStackLayoutDirective = __decorate([
        core_1.Directive({
            selector: "TKDataFormStackLayout"
        }),
        __param(0, core_1.Inject(core_1.ElementRef)),
        __metadata("design:paramtypes", [core_1.ElementRef])
    ], TKDataFormStackLayoutDirective);
    return TKDataFormStackLayoutDirective;
}());
exports.TKDataFormStackLayoutDirective = TKDataFormStackLayoutDirective;
var TKPropertyGroupDirective = (function () {
    function TKPropertyGroupDirective(_elementRef) {
        this._elementRef = _elementRef;
        this._propertyGroup = this._elementRef.nativeElement;
    }
    Object.defineProperty(TKPropertyGroupDirective.prototype, "propertyGroup", {
        get: function () {
            return this._propertyGroup;
        },
        enumerable: true,
        configurable: true
    });
    TKPropertyGroupDirective = __decorate([
        core_1.Directive({
            selector: "TKPropertyGroup"
        }),
        __param(0, core_1.Inject(core_1.ElementRef)),
        __metadata("design:paramtypes", [core_1.ElementRef])
    ], TKPropertyGroupDirective);
    return TKPropertyGroupDirective;
}());
exports.TKPropertyGroupDirective = TKPropertyGroupDirective;
var TKPropertyEditorDirective = (function () {
    function TKPropertyEditorDirective(_elementRef) {
        this._elementRef = _elementRef;
        this._propertyEditor = this._elementRef.nativeElement;
    }
    Object.defineProperty(TKPropertyEditorDirective.prototype, "propertyEditor", {
        get: function () {
            return this._propertyEditor;
        },
        enumerable: true,
        configurable: true
    });
    TKPropertyEditorDirective = __decorate([
        core_1.Directive({
            selector: "TKPropertyEditor"
        }),
        __param(0, core_1.Inject(core_1.ElementRef)),
        __metadata("design:paramtypes", [core_1.ElementRef])
    ], TKPropertyEditorDirective);
    return TKPropertyEditorDirective;
}());
exports.TKPropertyEditorDirective = TKPropertyEditorDirective;
var TKCustomPropertyEditorDirective = (function () {
    function TKCustomPropertyEditorDirective(_elementRef) {
        this._elementRef = _elementRef;
        this._propertyEditor = this._elementRef.nativeElement;
    }
    Object.defineProperty(TKCustomPropertyEditorDirective.prototype, "propertyEditor", {
        get: function () {
            return this._propertyEditor;
        },
        enumerable: true,
        configurable: true
    });
    TKCustomPropertyEditorDirective = __decorate([
        core_1.Directive({
            selector: "TKCustomPropertyEditor"
        }),
        __param(0, core_1.Inject(core_1.ElementRef)),
        __metadata("design:paramtypes", [core_1.ElementRef])
    ], TKCustomPropertyEditorDirective);
    return TKCustomPropertyEditorDirective;
}());
exports.TKCustomPropertyEditorDirective = TKCustomPropertyEditorDirective;
var TKDataFormPropertyDirective = (function () {
    function TKDataFormPropertyDirective(owner, _elementRef) {
        this.owner = owner;
        this._elementRef = _elementRef;
    }
    TKDataFormPropertyDirective.prototype.ngOnInit = function () {
        var property = this._elementRef.nativeElement;
        if (this.owner.dataForm.properties) {
            this.owner.dataForm.properties.push(property);
        }
        else {
            this.owner.dataForm.properties = new Array(property);
        }
    };
    TKDataFormPropertyDirective = __decorate([
        core_1.Directive({
            selector: "[tkDataFormProperty]"
        }),
        __param(0, core_1.Inject(RadDataFormComponent)),
        __param(1, core_1.Inject(core_1.ElementRef)),
        __metadata("design:paramtypes", [RadDataFormComponent,
            core_1.ElementRef])
    ], TKDataFormPropertyDirective);
    return TKDataFormPropertyDirective;
}());
exports.TKDataFormPropertyDirective = TKDataFormPropertyDirective;
var TKPropertyGroupLayoutDirective = (function () {
    function TKPropertyGroupLayoutDirective(owner, _elementRef) {
        this.owner = owner;
        this._elementRef = _elementRef;
    }
    TKPropertyGroupLayoutDirective.prototype.ngOnInit = function () {
        var layout = this._elementRef.nativeElement;
        this.owner.propertyGroup.layout = layout;
    };
    TKPropertyGroupLayoutDirective = __decorate([
        core_1.Directive({
            selector: "[tkPropertyGroupLayout]"
        }),
        __param(0, core_1.Inject(TKPropertyGroupDirective)),
        __param(1, core_1.Inject(core_1.ElementRef)),
        __metadata("design:paramtypes", [TKPropertyGroupDirective,
            core_1.ElementRef])
    ], TKPropertyGroupLayoutDirective);
    return TKPropertyGroupLayoutDirective;
}());
exports.TKPropertyGroupLayoutDirective = TKPropertyGroupLayoutDirective;
var TKDataFormGroupsDirective = (function () {
    function TKDataFormGroupsDirective(owner, _elementRef) {
        this.owner = owner;
        this._elementRef = _elementRef;
    }
    TKDataFormGroupsDirective.prototype.ngOnInit = function () {
        var property = this._elementRef.nativeElement;
        if (this.owner.dataForm.groups) {
            this.owner.dataForm.groups.push(property);
        }
        else {
            this.owner.dataForm.groups = new Array(property);
        }
    };
    TKDataFormGroupsDirective = __decorate([
        core_1.Directive({
            selector: "[tkDataFormGroups]"
        }),
        __param(0, core_1.Inject(RadDataFormComponent)),
        __param(1, core_1.Inject(core_1.ElementRef)),
        __metadata("design:paramtypes", [RadDataFormComponent,
            core_1.ElementRef])
    ], TKDataFormGroupsDirective);
    return TKDataFormGroupsDirective;
}());
exports.TKDataFormGroupsDirective = TKDataFormGroupsDirective;
var TKPropertyGroupTitleStyleDirective = (function () {
    function TKPropertyGroupTitleStyleDirective(owner, _elementRef) {
        this.owner = owner;
        this._elementRef = _elementRef;
    }
    TKPropertyGroupTitleStyleDirective.prototype.ngOnInit = function () {
        var titleStyle = this._elementRef.nativeElement;
        this.owner.propertyGroup.titleStyle = titleStyle;
    };
    TKPropertyGroupTitleStyleDirective = __decorate([
        core_1.Directive({
            selector: "[tkPropertyGroupTitleStyle]"
        }),
        __param(0, core_1.Inject(TKPropertyGroupDirective)),
        __param(1, core_1.Inject(core_1.ElementRef)),
        __metadata("design:paramtypes", [TKPropertyGroupDirective,
            core_1.ElementRef])
    ], TKPropertyGroupTitleStyleDirective);
    return TKPropertyGroupTitleStyleDirective;
}());
exports.TKPropertyGroupTitleStyleDirective = TKPropertyGroupTitleStyleDirective;
var TKPropertyEditorStyleDirective = (function () {
    function TKPropertyEditorStyleDirective(owner, _elementRef) {
        this.owner = owner;
        this._elementRef = _elementRef;
    }
    TKPropertyEditorStyleDirective.prototype.ngOnInit = function () {
        var style = this._elementRef.nativeElement;
        this.owner.propertyEditor.propertyEditorStyle = style;
    };
    TKPropertyEditorStyleDirective = __decorate([
        core_1.Directive({
            selector: "[tkPropertyEditorStyle]"
        }),
        __param(0, core_1.Inject(TKPropertyEditorDirective)),
        __param(1, core_1.Inject(core_1.ElementRef)),
        __metadata("design:paramtypes", [TKPropertyEditorDirective,
            core_1.ElementRef])
    ], TKPropertyEditorStyleDirective);
    return TKPropertyEditorStyleDirective;
}());
exports.TKPropertyEditorStyleDirective = TKPropertyEditorStyleDirective;
var TKCustomPropertyEditorStyleDirective = (function () {
    function TKCustomPropertyEditorStyleDirective(owner, _elementRef) {
        this.owner = owner;
        this._elementRef = _elementRef;
    }
    TKCustomPropertyEditorStyleDirective.prototype.ngOnInit = function () {
        var style = this._elementRef.nativeElement;
        this.owner.propertyEditor.propertyEditorStyle = style;
    };
    TKCustomPropertyEditorStyleDirective = __decorate([
        core_1.Directive({
            selector: "[tkCustomPropertyEditorStyle]"
        }),
        __param(0, core_1.Inject(TKCustomPropertyEditorDirective)),
        __param(1, core_1.Inject(core_1.ElementRef)),
        __metadata("design:paramtypes", [TKCustomPropertyEditorDirective,
            core_1.ElementRef])
    ], TKCustomPropertyEditorStyleDirective);
    return TKCustomPropertyEditorStyleDirective;
}());
exports.TKCustomPropertyEditorStyleDirective = TKCustomPropertyEditorStyleDirective;
var TKPropertyGroupPropertiesDirective = (function () {
    function TKPropertyGroupPropertiesDirective(owner, _elementRef) {
        this.owner = owner;
        this._elementRef = _elementRef;
    }
    TKPropertyGroupPropertiesDirective.prototype.ngOnInit = function () {
        var property = this._elementRef.nativeElement;
        if (this.owner.propertyGroup.properties) {
            this.owner.propertyGroup.properties.push(property);
        }
        else {
            this.owner.propertyGroup.properties = new Array(property);
        }
    };
    TKPropertyGroupPropertiesDirective = __decorate([
        core_1.Directive({
            selector: "[tkPropertyGroupProperties]"
        }),
        __param(0, core_1.Inject(TKPropertyGroupDirective)),
        __param(1, core_1.Inject(core_1.ElementRef)),
        __metadata("design:paramtypes", [TKPropertyGroupDirective,
            core_1.ElementRef])
    ], TKPropertyGroupPropertiesDirective);
    return TKPropertyGroupPropertiesDirective;
}());
exports.TKPropertyGroupPropertiesDirective = TKPropertyGroupPropertiesDirective;
var TKEntityPropertyEditorDirective = (function () {
    function TKEntityPropertyEditorDirective(owner, _elementRef) {
        this.owner = owner;
        this._elementRef = _elementRef;
    }
    TKEntityPropertyEditorDirective.prototype.ngOnInit = function () {
        var editor = this._elementRef.nativeElement;
        this.owner.entityProperty.editor = editor;
    };
    TKEntityPropertyEditorDirective = __decorate([
        core_1.Directive({
            selector: "[tkEntityPropertyEditor]"
        }),
        __param(0, core_1.Inject(TKEntityPropertyDirective)),
        __param(1, core_1.Inject(core_1.ElementRef)),
        __metadata("design:paramtypes", [TKEntityPropertyDirective,
            core_1.ElementRef])
    ], TKEntityPropertyEditorDirective);
    return TKEntityPropertyEditorDirective;
}());
exports.TKEntityPropertyEditorDirective = TKEntityPropertyEditorDirective;
var TKEntityPropertyValidatorsDirective = (function () {
    function TKEntityPropertyValidatorsDirective(owner, _elementRef) {
        this.owner = owner;
        this._elementRef = _elementRef;
    }
    TKEntityPropertyValidatorsDirective.prototype.ngOnInit = function () {
        var validator = this._elementRef.nativeElement;
        if (this.owner.entityProperty.validators) {
            this.owner.entityProperty.validators.push(validator);
        }
        else {
            this.owner.entityProperty.validators = new Array(validator);
        }
    };
    TKEntityPropertyValidatorsDirective = __decorate([
        core_1.Directive({
            selector: "[tkEntityPropertyValidators]"
        }),
        __param(0, core_1.Inject(TKEntityPropertyDirective)),
        __param(1, core_1.Inject(core_1.ElementRef)),
        __metadata("design:paramtypes", [TKEntityPropertyDirective,
            core_1.ElementRef])
    ], TKEntityPropertyValidatorsDirective);
    return TKEntityPropertyValidatorsDirective;
}());
exports.TKEntityPropertyValidatorsDirective = TKEntityPropertyValidatorsDirective;
exports.DATAFORM_DIRECTIVES = [RadDataFormComponent, TKDataFormPropertyDirective, TKEntityPropertyEditorDirective, TKEntityPropertyDirective, TKDataFormGroupsDirective, TKPropertyGroupPropertiesDirective, TKPropertyGroupDirective, TKPropertyGroupTitleStyleDirective, TKPropertyEditorDirective, TKPropertyEditorStyleDirective, TKEntityPropertyValidatorsDirective, TKDataFormGridLayoutDirective, TKDataFormStackLayoutDirective, TKPropertyGroupLayoutDirective, TKCustomPropertyEditorDirective, TKCustomPropertyEditorStyleDirective];
element_registry_1.registerElement("RadDataForm", function () { return _1.RadDataForm; });
element_registry_1.registerElement("TKEntityProperty", function () { return _1.EntityProperty; });
element_registry_1.registerElement("TKPropertyEditor", function () { return _1.PropertyEditor; });
element_registry_1.registerElement("TKCustomPropertyEditor", function () { return _1.CustomPropertyEditor; });
element_registry_1.registerElement("TKPropertyGroup", function () { return _1.PropertyGroup; });
element_registry_1.registerElement("TKGroupTitleStyle", function () { return _1.GroupTitleStyle; });
element_registry_1.registerElement("TKPropertyEditorStyle", function () { return _1.PropertyEditorStyle; });
element_registry_1.registerElement("TKPropertyValidator", function () { return _1.PropertyValidator; });
element_registry_1.registerElement("TKNonEmptyValidator", function () { return _1.NonEmptyValidator; });
element_registry_1.registerElement("TKMaximumLengthValidator", function () { return _1.MaximumLengthValidator; });
element_registry_1.registerElement("TKMinimumLengthValidator", function () { return _1.MinimumLengthValidator; });
element_registry_1.registerElement("TKEmailValidator", function () { return _1.EmailValidator; });
element_registry_1.registerElement("TKRangeValidator", function () { return _1.RangeValidator; });
element_registry_1.registerElement("TKPhoneValidator", function () { return _1.PhoneValidator; });
element_registry_1.registerElement("TKIsTrueValidator", function () { return _1.IsTrueValidator; });
element_registry_1.registerElement("TKRegExValidator", function () { return _1.RegExValidator; });
element_registry_1.registerElement("TKDataFormGridLayout", function () { return _1.DataFormGridLayout; });
element_registry_1.registerElement("TKDataFormStackLayout", function () { return _1.DataFormStackLayout; });
var NativeScriptUIDataFormModule = (function () {
    function NativeScriptUIDataFormModule() {
    }
    NativeScriptUIDataFormModule = __decorate([
        core_1.NgModule({
            declarations: [exports.DATAFORM_DIRECTIVES],
            exports: [exports.DATAFORM_DIRECTIVES]
        })
    ], NativeScriptUIDataFormModule);
    return NativeScriptUIDataFormModule;
}());
exports.NativeScriptUIDataFormModule = NativeScriptUIDataFormModule;
